src/Symfony/Component/Notifier/Bridge/Smsbox/SmsboxTransport.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Smsbox;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Alan Zarli <azarli@smsbox.fr>
+ * @author Farid Touil <ftouil@smsbox.fr>
+ */
+final class SmsboxTransport extends AbstractTransport
+{
+    protected const HOST = 'api.smsbox.pro';
+
+    private string $apiKey;
+    private ?string $mode;
+    private ?int $strategy;
+    private ?string $sender;
+
+    public function __construct(#[\SensitiveParameter] string $apiKey, string $mode, int $strategy, ?string $sender, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->apiKey = $apiKey;
+        $this->mode = $mode;
+        $this->strategy = $strategy;
+        $this->sender = $sender;
+
+        SmsboxOptions::validMode($this->mode);
+        SmsboxOptions::validStrategy($this->strategy);
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (SmsboxOptions::MESSAGE_MODE_EXPERT === $this->mode) {
+            return sprintf('smsbox://%s?mode=%s&strategy=%s&sender=%s', $this->getEndpoint(), $this->mode, $this->strategy, $this->sender);
```

Comment:
Is `sender` only supported if mode is expert? 🤔

Reply:
Yes, custom sender ID is only available by specifying "mode =Expert" on our API interface.

Reply:
Is this a paid extra?

Reply:
For the vast majority of destinations, custom sender ID is free of charge. For some destinations, local regulations require prior declaration and some operators charge registration fees.